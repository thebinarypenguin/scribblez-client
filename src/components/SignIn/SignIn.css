#SignIn .dialog {
  font-size: 1.5rem;
  text-align: center;
  padding-top: 4rem;
  padding-bottom: 4rem;
}

#SignIn h2 {
  margin-bottom: 2rem;
}


#SignIn input {
  display: block;
  margin: auto;
  width: 20rem;
  padding: 3px 6px;
  margin-bottom: 1rem;
}


#SignIn button {
  padding: 3px 6px;
}

#demo-credentials {
  list-style-type: none;
  width: 20rem;
  margin: auto;
  font-size: 1rem;
  margin-bottom: 2rem;
  text-align: left;
}
