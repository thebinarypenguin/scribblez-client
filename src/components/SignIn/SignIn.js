import React from 'react';
import UserContext from '../../contexts/UserContext';
import AuthService from '../../services/AuthService';
import TopBar from '../../components/TopBar/TopBar';
import Utils from '../../services/Utils';

import './SignIn.css';

class SignIn extends React.Component {

  static contextType = UserContext;

  constructor(props) {
    super(props);

    this.state = {
      error: null,
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.generateLinks = this.generateLinks.bind(this);
  }

  generateLinks() {

    let links = [];

    if (Utils.isTokenActive(window.localStorage.getItem('token'))) {

      links = [
        { name: 'Feed',     target: '/feed'     },
        { name: 'Notes',    target: '/notes'    },
        { name: 'Sign Out', target: '/sign-out' },
      ];

    } else {

      links = [
        { name: 'Feed',    target: '/feed'    },
        { name: 'Sign Up', target: '/sign-up' },
        { name: 'Sign In', target: '/sign-in' },
      ];
    }

    return links;
  }

  handleSubmit(ev) {

    ev.preventDefault();

    const { username, password } = ev.target;

    // console.log('IOU form validation');

    return AuthService
      .login(username.value, password.value)
      .then((resp) => {
        this.context.logIn(resp.token);
        this.props.history.push('/notes');
      })
      .catch((err) => {
        this.setState({ error: err.message });
      });
  }

  render() {

    return (
      <div id="SignIn">
        <TopBar links={this.generateLinks()} />

        <div className="dialog">

        <h2>Sign In</h2>

        <ul id="demo-credentials">
          <li>Demo username: <strong>homer</strong></li>
          <li>Demo password: <strong>password</strong></li>
        </ul>

        <form autoComplete="off" onSubmit={this.handleSubmit} >

          <input type="text" id="username" placeholder="Username" />
          <input type="password" id="password" placeholder="Password" />
          <button type="submit">Sign In</button>

        </form>

      </div>

    </div>
    );
  }
}

export default SignIn;
