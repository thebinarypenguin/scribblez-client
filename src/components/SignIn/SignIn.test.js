import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import SignIn from './SignIn';

it('renders without crashing', () => {

  const container = document.createElement('div');

  const subject = (
    <MemoryRouter>
      <SignIn />
    </MemoryRouter>
  );

  ReactDOM.render(subject, container);
  ReactDOM.unmountComponentAtNode(container);
});

it('renders the UI as expected', () => {

  const subject = (
    <MemoryRouter>
      <SignIn />
    </MemoryRouter>
  );

  const tree = renderer.create(subject).toJSON();
  expect(tree).toMatchSnapshot();
});
