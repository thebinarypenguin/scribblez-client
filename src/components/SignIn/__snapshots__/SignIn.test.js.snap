// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  id="SignIn"
>
  <div
    id="TopBar"
  >
    <div
      className="Container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1>
          Scribblez
        </h1>
      </a>
      <ul>
        <li>
          <a
            href="/feed"
            onClick={[Function]}
          >
            Feed
          </a>
        </li>
        <li>
          <a
            href="/sign-up"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </li>
        <li>
          <a
            href="/sign-in"
            onClick={[Function]}
          >
            Sign In
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="dialog"
  >
    <h2>
      Sign In
    </h2>
    <ul
      id="demo-credentials"
    >
      <li>
        Demo username: 
        <strong>
          homer
        </strong>
      </li>
      <li>
        Demo password: 
        <strong>
          password
        </strong>
      </li>
    </ul>
    <form
      autoComplete="off"
      onSubmit={[Function]}
    >
      <input
        id="username"
        placeholder="Username"
        type="text"
      />
      <input
        id="password"
        placeholder="Password"
        type="password"
      />
      <button
        type="submit"
      >
        Sign In
      </button>
    </form>
  </div>
</div>
`;
